src/ripple/core/impl/Workers.cpp

```diff
@@ -144,27 +143,50 @@ void Workers::deleteWorkers (beast::LockFreeStack <Worker>& stack)
 //------------------------------------------------------------------------------
 
 Workers::Worker::Worker (Workers& workers, std::string const& threadName)
-    : Thread (threadName)
-    , m_workers (workers)
+    : m_workers {workers}
+    , threadName_ {threadName}
+    , wakeCount_ {0}
+    , shouldExit_ {false}
 {
-    startThread ();
+    thread_ = std::thread {&Workers::Worker::run, this};
 }
 
 Workers::Worker::~Worker ()
 {
-    stopThread ();
+    {
+        std::lock_guard <std::mutex> lock {mutex_};
+        ++wakeCount_;
+        shouldExit_ = true;
+    }
+
+    wakeup_.notify_one();
+    thread_.join();
+}
+
+void Workers::Worker::notify ()
+{
+    {
+        std::lock_guard <std::mutex> lock {mutex_};
+        ++wakeCount_;
+    }
+    wakeup_.notify_one();
```

Comment:
I guess with Scott D's link, the question is:  Can one thread call `Worker::notify()` while another thread is calling `Worker::~Worker()`.  If not, then no problems.  But if so, `Worker::~Worker()` might begin and end between lines 171 and 172, causing line 172 to call `notify_one()` after `wakeup_` has destructed.